File: src/main/java/com/app/core/model/DmHolder.java
package com.app.core.model;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.List;

public class DmHolder<T> implements DmBase {
    private List<T> lst;
    private String fn;
    private Object payLoad;
    private int index;

    public DmHolder(String fn) {
        this.fn = fn;
    }

    public DmHolder(@NonNull String fn, List<T> value) {
        this.lst = value;
        this.fn = fn;
    }

    public DmHolder(@NonNull String fn, int index, @Nullable Object payLoad) {
        this.fn = fn;
        this.payLoad = payLoad;
        this.index = index;
    }

    public Object getPayLoad() {
        return payLoad;
    }

    public int getIndex() {
        return index;
    }

    public List<T> getLst() {
        return lst;
    }

    public String getFn() {
        return fn;
    }
}

File: src/main/java/com/app/core/ui/recycler/DmHolder.kt
package com.app.core.ui.recycler

import com.app.core.model.DmBase

class DmHolder<T> : DmBase {
    var lst: List<T?>? = null
    var fn: String? = null
    var payLoad: Any? = null
    var index: Int = -1

    constructor(fn: String) {
        this.fn = fn
    }

    constructor(fn: String?, value: List<T?>?) {
        this.lst = value
        this.fn = fn
    }

    constructor(fn: String?, index: Int, payLoad: Any?) {
        this.fn = fn
        this.payLoad = payLoad
        this.index = index
    }
}
